src/service-worker.js

```diff
@@ -0,0 +1,421 @@
+    });
+};
+
+/**
+ * passThrough
+ * Inspect event.request to determine whether it's an okapi request.
+ * If it is, make sure its AT is valid or perform RTR before executing it.
+ * If it isn't, execute it immediately. If RTR fails catastrophically,
+ * post an RTR_ERROR message to clients and return an empty Response in a
+ * resolving promise in order to let the top-level error-handler pick it up.
+ * @param {Event} event
+ * @param {object} te tokenExpiration shaped like { atExpires, rtExpires }
+ * @param {string} oUrl okapiUrl
+ * @returns Promise
+ * @throws if any fetch fails
+ */
+export const passThrough = (event, te, oUrl) => {
+  const req = event.request.clone();
+
+  // okapi requests are subject to RTR
+  if (isOkapiRequest(req, oUrl)) {
+    // console.log('-- (rtr-sw) => will fetch', req.url);
+    if (isLogoutRequest(req, oUrl)) {
+      return passThroughLogout(event);
+    }
+
+    if (isPermissibleRequest(req, te, oUrl)) {
+      return passThroughWithAT(event);
+    }
+
+    if (isValidRT(te)) {
+      // console.log('-- (rtr-sw) =>      valid RT');
+      return passThroughWithRT(event);
+    }
+
+    // kill me softly: send an empty response body, which allows the fetch
+    // to return without error while the clients catch up, read the RTR_ERROR
+    // and handle it, hopefully by logging out.
+    // Promise.reject() here would result in every single fetch in every
+    // single application needing to thoughtfully handle RTR_ERROR responses.
+    messageToClient(event, { type: 'RTR_ERROR', error: 'AT/RT failure' });
+    return Promise.resolve(new Response({}));
+  }
+
+  // default: pass requests through to the network
+  // console.log('-- (rtr-sw) passThrough NON-OKAPI', req.url)
+  return fetch(event.request, { credentials: 'include' })
+    .catch(e => {
+      console.error(e); // eslint-disable-line no-console
+      return Promise.reject(new Error(e));
+    });
+};
+
+/**
+ * install
+ * on install, force this SW to be the active SW
+ */
+self.addEventListener('install', (_event) => {
+  // console.log('-- (rtr-sw) => install', event);
+  return self.skipWaiting();
+});
+
+/**
+ * activate
+ * on activate, force this SW to control all in-scope clients,
+ * even those that loaded before this SW was registered.
+ */
+self.addEventListener('activate', async (event) => {
+  // console.log('-- (rtr-sw) => activate', event);
+  event.waitUntil(self.clients.claim());
+});
+
+/**
+ * eventListener: message
+ * listen for messages from @folio/stripes-core clients and dispatch them accordingly.
+ */
+self.addEventListener('message', async (event) => {
+  if (event.data.source === '@folio/stripes-core') {
+    console.info('-- (rtr-sw) reading', event.data);
+    if (event.data.type === 'OKAPI_CONFIG') {
+      okapiUrl = event.data.value.url;
+      okapiTenant = event.data.value.tenant;
+    }
+
+    // for reasons unclear to me, this does not work. the value comes through
```

Comment:
What doesn't work?

Reply:
Ok I get that what doesn't work is the whole point of the comment. But how is what is not working related to the goal of the PR?

Reply:
I was trying to leverage the stripes logger within the service worker so we could have access to the same filter tools but it doesn't work: the logger object gets munged on its way into the service worker. But don't worry about this. It's tangential to the core work here, and will probably just be removed before it's committed.